api.d.ts

```diff
@@ -1113,8 +1113,56 @@ export interface ILexerConfig {
      * Note that this would have negative performance implications.
      */
     safeMode?: boolean
+
+    /**
+     * A custom error message provider.
+     * Can be used to override the default error messages.
+     * For example:
+     *   - Translating the error messages to a different languages.
+     *   - Changing the formatting.
+     */
+    errorMessageProvider?: ILexerErrorMessageProvider
 }
 
+export interface ILexerErrorMessageProvider {
+    /**
+     * Unexpected Character Error happens when lexer could not find token match for input text
+     *
+     * @param text - Input text.
+     *
+     * @param errorStartOffset - Offset in input text where error starts.
+     *
+     * @param errorLength - Error length.
+     *
+     * @param errorLine - Line number where the error occured.
+     *
+     * @param errorColumn - Column number where the error occured.
+     */
+    buildUnexpectedCharacterMessage(
+        text: string,
+        errorStartOffset: number,
+        errorLength: number,
+        errorLine: number,
+        errorColumn: number
+    ): string
+
+    /**
+     * Unable To Pop Lexer Mode Error happens when lexer tries to pop the last remaining mode from the mode stack.
+     *
```

Comment:
- The description needs to indicate that this is only relevant for multi-mode lexers.
- The description could include a suffix that explains that the situation described means that
   "there is no longer any active lexer mode"